lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOAccessApprovalDAOImpl.java

```diff
@@ -517,7 +523,34 @@ public List<AccessApproval> searchAccessApprovals(String accessorId, String acce
 		
 		return jdbcTemplate.query(sql, ROW_MAPPER, queryParams.toArray());
 	}
-	
+
+	@Override
+	public Map<Long, AccessApproval> searchAccessApprovalsForSubmission(Set<Long> submissionIDs, String accessorId) {
+		ValidateArgument.requiredNotEmpty(submissionIDs, "submissionIDs");
+		ValidateArgument.required(accessorId, "accessorId");
+
+		String sql = "SELECT A.*, S.ID AS SUBMISSION_ID FROM " + TABLE_ACCESS_APPROVAL +
+				" A JOIN " + TABLE_DATA_ACCESS_SUBMISSION + " S On S." + CREATED_BY + " = A."
+				+ COL_DATA_ACCESS_SUBMISSION_SUBMITTER_SUBMITTER_ID + " and S." +
+				COL_DATA_ACCESS_SUBMISSION_ACCESS_REQUIREMENT_ID + " = A." + COL_ACCESS_APPROVAL_REQUIREMENT_ID +
```

Comment:
You need to join on the version too? If you don't, you might have multiple approval for the same user and submitter with different requirement versions and this method will fail. But since one approval for a given requirement is independent of the version you could sort by version desc and take the first one.

We need a test for this scenario, it looks like it would be possible to have multiple access approvals for the same requirement and submitter/accessor if the there are multiple versions of the requirement.